pontoon/base/models.py

```diff
@@ -1401,6 +1391,37 @@ def can_commit(self):
         """True if we can commit strings back to this repo."""
         return self.type in ('svn', 'git', 'hg')
 
+    @property
+    def api_config(self):
+        """
+        Repository API configuration consists of:
+        - Endpoint: A URL pattern to get repository metadata from. Used during sync for faster
+          retrieval of latest commit hashes in combination with the Key.
+          Supports {locale_code} wildcard.
+        - Key: A string used to retrieve the latest commit hash from the JSON response.
+        """
+        url = self.url
+
+        if url.startswith('ssh://hg.mozilla.org/'):
+            endpoint_base = url.replace('ssh://', 'https://')
```

Comment:
@Pike :+1: Optionally, we could do something like this:
```
url = urlparse(self.url)
# That dict can be moved into constant somewhere above.
supported_repos =  {
  'hg.mozilla.org': {
       'endpoint': 'https://hg.mozilla.org/json-rev/default',
      'node': 'node',
   },
  (...)
}
if url.netloc in supported_repos:
   repo = supported_repos[url.netloc]
   repo['endpoint'] = repo['endpoint'].format(url)
   return repo
else:
   return None
```
That could reduce number of nested code blocks in this property in the future and even we could make this more reusable for non-mozilla related projects (?) <insert philosophical raptor here>